### PR TITLE
Add DEX helm chart to Kuberpult helm chart 

### PR DESCRIPTION
**Pull request by bernardo-bastos**
If Dex is enabled, installs the DEX helm chart together with Kuberpult.

REV: DSN-0Z3L73

### CODE REVIEW

File: charts/kuberpult/Chart.yaml.tpl [hunk 1]
```diff
@@ -23,6 +23,16 @@ version: $VERSION
 # It is recommended to use it with quotes.
 appVersion: "$VERSION"
 
+# This is is the DEX helm chart only to installed it RBAC is enabled.
```
Comment (sven-urbanski-freiheit-com) [inline charts/kuberpult/Chart.yaml.tpl:26]:
```suggestion
# This is the DEX helm chart which will only be installed it RBAC is enabled.
```

File: charts/kuberpult/Chart.yaml.tpl [hunk 2]
```diff
@@ -23,6 +23,16 @@ version: $VERSION
...
+# Dex is an identity service that uses OpenID Connect to drive authentication through other 
+# identity providers.  
+# For more information please check: https://github.com/dexidp/dex
+dependencies:
+- name: dex
+condition: dex.enabled
+version: "0.14.2"
```
Comment (sven-urbanski-freiheit-com) [inline charts/kuberpult/Chart.yaml.tpl:33]: Can we make the version configurable?

File: charts/kuberpult/values.yaml.tpl [hunk 3]
```diff
@@ -218,3 +218,26 @@ dex:
   baseURL: ""
   # List of scopes to validate the token. Please add them as comma separated values.
   scopes: ""
+  # List of environment variables to be added to the dex service pod.
```
Comment (sven-urbanski-freiheit-com) [inline charts/kuberpult/values.yaml.tpl:221]: @hannesg please review this PR too

File: charts/kuberpult/Chart.yaml.tpl (hunk 2 above)
Comment (hannesg) [reply charts/kuberpult/Chart.yaml.tpl:33]: :thinking: Maybe also make it possible to use an external dex?

**Review by hannesg — COMMENTED** [review]
I think this is tricky because you are re-using the `dex` namespace in the values. I think it would be better to move our own config to a seperate top-level value like `auth.dex` to seperate concerns. But apart from that, I'm fine.

File: charts/kuberpult/values.yaml.tpl [hunk 4]
```diff
@@ -199,22 +199,48 @@ auth:
...
+    # List of environment variables to be added to the dex service pod.
+    # Example, if you want your DEX service to have access to to the OAUTH_CLIENT_ID, you can specify
+    # it the following way: 
+    # 
+    #  - name: OAUTH_CLIENT_ID
+    #    valueFrom:
+    #      secretKeyRef:
+    #      name: kuberpult-oauth-client-id
+    #        key: kuberpult-oauth-client-id
+    envVars: []
```
Comment (hannesg) [inline charts/kuberpult/values.yaml.tpl:231]: This config must be under `dex`, right? Otherwise the dex chart won't pick it up.

Comment (bernardo-bastos) [reply charts/kuberpult/values.yaml.tpl]: Yes, you are right. Made the necessary changes and re tested everything locally on my team dummy cluster.

File: charts/kuberpult/Chart.yaml.tpl (hunk 1 above)
Comment (bernardo-bastos) [reply charts/kuberpult/Chart.yaml.tpl]: Corrected the comment.

File: charts/kuberpult/Chart.yaml.tpl (hunk 2 above)
Comment (bernardo-bastos) [reply charts/kuberpult/Chart.yaml.tpl]: Added an option to omit Dex installation

**Review by hannesg — APPROVED** [review]

**Pull request merged**